net/src/ipv6/mod.rs

```diff
@@ -422,24 +422,58 @@ impl From<Ipv6ExtNext> for Header {
 
 #[cfg(any(test, feature = "arbitrary"))]
 mod contract {
+    use crate::ip::NextHeader;
     use crate::ipv6::Ipv6;
-    use bolero::{Driver, TypeGenerator};
+    use bolero::{Driver, TypeGenerator, ValueGenerator};
     use etherparse::Ipv6Header;
     use std::net::Ipv6Addr;
 
-    impl TypeGenerator for Ipv6 {
-        fn generate<D: Driver>(u: &mut D) -> Option<Self> {
+    /// A [`bolero::TypeGenerator`] for common (and supported) [`NextHeader`] values
+    #[derive(Copy, Clone, Debug, bolero::TypeGenerator)]
+    pub enum CommonNextHeader {
+        /// TCP next header (see [`NextHeader::TCP`]
+        Tcp,
+        /// UDP next header (see [`NextHeader::UDP`]
+        Udp,
+        /// ICMP v6 next header (see [`NextHeader::ICMP6`]
+        Icmp6,
+    }
+
+    impl From<CommonNextHeader> for NextHeader {
+        fn from(value: CommonNextHeader) -> Self {
+            match value {
+                CommonNextHeader::Tcp => NextHeader::TCP,
+                CommonNextHeader::Udp => NextHeader::UDP,
+                CommonNextHeader::Icmp6 => NextHeader::ICMP6,
+            }
+        }
+    }
+
+    /// [`ValueGenerator`] for an (otherwise) arbitrary [`Ipv6`] with a specified [`NextHeader`].
+    pub struct GenWithNextHeader(pub NextHeader);
+
+    impl ValueGenerator for GenWithNextHeader {
+        type Output = Ipv6;
+
+        fn generate<D: Driver>(&self, u: &mut D) -> Option<Ipv6> {
             let mut header = Ipv6(Ipv6Header::default());
             header
                 .set_source(u.gen()?)
                 .set_destination(Ipv6Addr::from(u.gen::<u128>()?))
-                .set_next_header(u.gen()?)
+                .set_next_header(self.0)
```

Comment:
Why is this safe for IPv6, but not for IPv4?

Reply:
That is a minor bug I will fix in a future PR.  Good point tho